app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -111,9 +112,27 @@ class BrowserWebViewClient(
     @WorkerThread
     override fun shouldInterceptRequest(webView: WebView, request: WebResourceRequest): WebResourceResponse? {
         return runBlocking {
+
             val documentUrl = withContext(Dispatchers.Main) { webView.url }
             Timber.v("Intercepting resource ${request.url} on page $documentUrl")
-            requestInterceptor.shouldIntercept(request, webView, documentUrl, webViewClientListener)
+            val interceptionAction = requestInterceptor.shouldIntercept(request, webView, documentUrl, webViewClientListener)
+
+            if (interceptionAction is InterceptAction.BlockAndLoadAlternativeUrl) {
```

Comment:
We have a when statement to process the return statement already, feels like a cognitive disconnect to also have an if statement for this special actions - I have to think about how we are handling `InterceptAction.BlockAndLoadAlternativeUrl` in two places. Why not combine it all in a when? That was we have one block of logic to handle each type rather than splitting it out in multiple places.